solr/modules/opentelemetry/src/test/org/apache/solr/opentelemetry/CustomTestOtelTracerConfigurator.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.opentelemetry;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import org.apache.solr.common.util.NamedList;
+
+public class CustomTestOtelTracerConfigurator extends OtelTracerConfigurator {
+
+  static {
+    if (System.getProperty("host") == null) {
+      System.setProperty("host", "localhost");
+    }
+  }
+
+  private static final InMemorySpanExporter exporter = InMemorySpanExporter.create();
```

Comment:
yes the error is strange. I don't think the exported is the cause but I added shutdown to it, it was a good suggestion (I think this gets closed via the otel sdk clode method).
the issue seems to be some component calls GlobalOpenTelemetry.get() before the actual init happens (as an aside this is a very strange design decision on otel side: calling GlobalOpenTelemetry.get() will actually _set_ it to noop and will not allow anyone else to set it again hence the failure in the logs). I am also considering the test itself as a culprit (it does a call to .get() to verify otel is active). still massaging the test code to make it into a test failure instead of a crash-like situation.

Reply:
updated the test to not crash solr and reflect the underlying issue. it does not make it less flaky, the more I look at this the more I worry about the very bad design of GlobalOpenTelemetry. I will need to keep a close eye on the error rates of this test going forward.

Reply:
ok. I think I found it. pushed a change to tests to force init tracer before the cluster init. this way there is no race between test side and cluster side. I am no longer seeing failures locally, running a long beast 50 times now.